src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCheck.java

```diff
@@ -61,6 +61,48 @@
  * <pre>
  * &lt;module name=&quot;NoWhitespaceBefore&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * int foo;
+ * foo ++; // violation, whitespace before '++' is not allowed
+ * foo++; // OK
+ * for (int i = 0 ; i &lt; 5; i++) {}  // violation
+ *            // ^ whitespace before ';' is not allowed
+ * for (int i = 0; i &lt; 5; i++) {} // OK
+ * int[][] array = { { 1, 2, 3 }
+ *                 , { 4, 5, 6 }    // violation,
+ *                 , { 7, 8, 9 } }; // violation, whitespace before ',' is not allowed
+ * int[][] array2 = { { 1, 2, 3 },
+ *                    { 4, 5, 6 },
+ *                    { 7, 8, 9 } }; // OK
+ * Customer c = new Customer(); // Customer is a user-defined class
```

Comment:
to avoid this comment please use chaining method based on jdk classes.
Here are examples on iterating over String https://www.techiedelight.com/iterate-over-characters-string-java/
like:
```
    "example".chars()
        .mapToObj(i -> Character.valueOf((char) i))
        .forEach(System.out::print);
```

Reply:
> Here are examples on iterating over String https://www.techiedelight.com/iterate-over-characters-string-java/

Thank You @romani ! That Information was Goldmine.